themes/default/content/blog/mlops-huggingface-llm-aws-sagemaker-python/index.md

```diff
@@ -0,0 +1,131 @@
+---
+date: 2023-09-08
+title: "Deploy a Huggingface Model on AWS SageMaker with Pulumi Python IaC"
+allow_long_title: true
+meta_desc: "Guided short tutorial on starting a Pulumi Infrastructure as Code project to deploy Huggingface LLM Models on Amazon Web Services SageMaker Machine Learning Platform"
+meta_image: "meta.png"
+authors:
+    - kat-morgan
+tags:
+    - ai
+    - ml
+    - iac
+    - aws
+    - llm
+    - vllm
+    - mlops
+    - llama
+    - llama2
+    - python
+    - sagemaker
+    - huggingface
+---
+
+Welcome to another post in the Pulumi + Python MLOps #MLOpsChallenge series!
+
+In this short tutorial we will deploy a publicly available Huggingface model on Amazon AWS SageMaker, and test a prompt using a short python script.
+
+We will be deploying:
+
+* The [NousResearch/Llama-2-7b-chat-hf](https://huggingface.co/NousResearch/Llama-2-7b-chat-hf) model on [Hugging Face](https://huggingface.co/) 
+* [Amazon AWS IAM Roles](https://www.pulumi.com/registry/packages/aws/api-docs/iam/role/)
+* [Amazon AWS SageMaker Model Endpoint](https://www.pulumi.com/registry/packages/aws/api-docs/sagemaker/model/)
+* [Amazon AWS CloudWatch MetricAlarm](https://www.pulumi.com/registry/packages/aws/api-docs/cloudwatch/metricalarm/)
+
+[Pulumi templates](../how-to-create-and-share-a-pulumi-template) enhance Infrastructure as Code (IaC) development by providing a pre-written code framework designed to help you bootstrap new pulumi projects quickly.
+
+You can use the `sagemaker-aws-python` template as a working python starting point for your own AWS SageMaker deployments, and customize the model you use, SageMaker configuration, and CloudWatch integration just to scratch the surface. Your only limits are your imagination.
+
+## Requirements
+
+* [Python3 (3.9+)](https://www.python.org/downloads/)
+* [Pulumi CLI](https://www.pulumi.com/docs/install/)
+* [Pulumi account](https://app.pulumi.com/signup)
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
+* [Configured & Logged in with AWS Credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-authentication.html)
+
+## Instructions
+
+### 1. Login to Pulumi Cloud and initialize stack
+
+Let's begin by logging into Pulumi Cloud
+
+```bash
+# There are many ways to store Pulumi state, here we use Pulumi Cloud
+# Other state backends include S3, local file, and more
+pulumi login
+```
+
+### 2. Prepare a new Pulumi Project
```

Comment:
isn't there a `pulumi config set aws:region <REGION>` somewhere in here. Yours goes direct to us-east-1, but I think you have to set it first right?

Reply:
it's a default provided by the template, user's are prompted to accept the default or override with their desired region when creating the template.